### PR TITLE
fix: Widen range for React types to include React 19

### PR DESCRIPTION
**Pull request by Artur-**
This allows using @lit/react with React 19

For #4857

### CODE REVIEW

**Comment by kyubisation** [comment]
Should this also widen the dependency range in the ssr-react package?
https://github.com/lit/lit/blob/main/packages/labs/ssr-react/package.json#L54-L55

**Comment by OvidijusParsiunas** [comment]
Hi, is there an estimated date when this will be reviewed? Thanks!

**Comment by Artur-** [comment]
> Should this also widen the dependency range in the ssr-react package?

Looks like it should, I added it.

**Comment by Artur-** [comment]
Is @justinfagnani still working on this project or is it somebody else?

File: .changeset/rare-cherries-bake.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@lit/react": patch
```
Comment (kyubisation) [inline .changeset/rare-cherries-bake.md:2]: With the ssr react package also being updated, add it to the changeset

**Comment by kyubisation** [comment]
I'll bring it up during the community call on Thursday.

File: .changeset/rare-cherries-bake.md (hunk 1 above)
Comment (Artur-) [reply .changeset/rare-cherries-bake.md:2]: Done, I think :)

Comment (kyubisation) [reply .changeset/rare-cherries-bake.md:2]: Thanks 👍 
I think you can remove the original changeset, as a single changset file should be enough.

**Review by sorvell — APPROVED** [review]

File: package-lock.json [hunk 2]
```diff
@@ -31843,7 +31843,7 @@
         "react-dom": "^18.2.0"
       },
       "peerDependencies": {
-        "@types/react": "17 || 18"
+        "@types/react": "17 || 18 || 19"
```
Comment (justinfagnani) [inline package-lock.json:31846]: I would expect a few more changes here. Does the package lock need to be regenerated?

Comment (Artur-) [reply package-lock.json:31846]: The lock change for ssr-react was apparently missing, added what "npm install" updates

**Review by justinfagnani — APPROVED** [review]

**Review by augustjk — APPROVED** [review]
Thanks! I do want to explore detect React version and passing props directly to React for >= 19 as described in https://github.com/lit/lit/issues/4857#issuecomment-2524849452 but that can be done as a separate improvement.

**Pull request merged**